service/src/main/java/com/ge/predix/acs/monitoring/AcsHealthAggregator.java

```
@@ -0,0 +1,100 @@
+package com.ge.predix.acs.monitoring;
+
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthAggregator;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+// Modified version of org.springframework.boot.actuate.health.OrderedHealthAggregator
+public class AcsHealthAggregator implements HealthAggregator {
+
+    static final Status DEGRADED_STATUS = new Status(AcsMonitoringUtilities.HealthCode.DEGRADED.toString());
+
+    private List<String> statusOrder;
+
+    AcsHealthAggregator() {
+        this.setStatusOrder(Status.DOWN, Status.OUT_OF_SERVICE, DEGRADED_STATUS, Status.UP, Status.UNKNOWN);
+    }
+
+    void setStatusOrder(final Status... statusOrder) {
+        String[] order = new String[statusOrder.length];
+        for (int i = 0; i < statusOrder.length; ++i) {
+            order[i] = statusOrder[i].getCode();
+        }
+        this.setStatusOrder(Arrays.asList(order));
+    }
+
+    void setStatusOrder(final List<String> statusOrder) {
+        Assert.notNull(statusOrder, "StatusOrder must not be null");
```

I believe we should throw an exception here, assert is good for tests i believe